app/src/main/java/school/cactus/cactusnotesmvvmsample/EmailValidator.kt

```diff
@@ -1,12 +1,16 @@
 package school.cactus.cactusnotesmvvmsample
 
-class EmailValidator {
+import android.content.Context
+import android.content.res.Resources
+
+class EmailValidator{
+
     fun validate(email: String?) = when {
-        email == null -> "Email cannot be empty"
-        email.isBlank() -> "Email cannot be empty"
-        email.length < 5 -> "Email cannot be this short"
+        email == null -> R.string.email_error_msg_1
```

Comment:
It's better to name string resources with names that better represent their contents.

So instead of `R.string.email_error_msg_1`, I would use `R.string.email_cannot_be_empty`